Tools/automation.ps1

```diff
@@ -0,0 +1,790 @@
+            $Object.Parameters | ConvertTo-Json -Depth 7 | Out-File -FilePath $Object.ParameterPath -Force
+        }
+    }
+    End
+    {
+        ## Returns the completed object.
+        Return $ARMObjects
+    }
+}
+
+Function Test-ARMTemplate
+{
+    param(
+        [Parameter(Position=0)] $ARMObjects
+    )
+    Begin
+    {
+        Write-Host "`n"
+    }
+    Process
+    {
+        Write-Host "Verifying the ARM Resource Templates and Parameters are valid:"
+        $TestResults = ""
+
+        ## Parses through all ARM Parameter objects to validate they are properly configured.
+        Foreach($Object in $ARMObjects)
+        {
+            ## Validates that each ARM object will work.
+            Write-Host "  Validation testing on ARM Resource ($($Object.ObjectType))..."
+            $Result = Test-AzResourceGroupDeployment -ResourceGroupName $AzResourceGroup -Mode Complete -TemplateFile $Object.TemplatePath -TemplateParameterFile $Object.ParameterPath
+
+            ## If the ARM object fails validation, report error to screen.
+            IF($Result -ne "")
+            { 
+                ## Testing fails.
+                Write-Host "    ERROR:  $Result" -ForegroundColor Red
+                $TestResults += "$($Object.ObjectType):`n$Result`n`n"
+            }
+        }
+    }
+    End
+    {
+        ## Returns the TestResults.
+        Return $TestResults
+    }
+}
+
+Function New-ARMObjects
+{
+    param(
+        [Parameter(Position=0)] $ARMObjects,
+        [Parameter(Position=1)] [string] $ArchiveFunctionZip
+    )
+    Begin
+    {
+        Write-Host "`n"
+
+        ## Path to the compiled Functions compressed into a Zip file for upload to Azure Function
+        #$ArchiveFunctionZip = "$WorkingDirectory\CompiledFunctions.zip"
+        
+        ## Imports the contents of the Parameter Files for reference and logging purposes:
+        $jsonStorageAccount = Get-Content -Path $($ARMObjects.Where({$_.ObjectType -eq "StorageAccount"}).ParameterPath) | ConvertFrom-Json
+        $jsonKeyVault       = Get-Content -Path $($ARMObjects.Where({$_.ObjectType -eq "Keyvault"}).ParameterPath) | ConvertFrom-Json
+        $jsonFunction       = Get-Content -Path $($ARMObjects.Where({$_.ObjectType -eq "Function"}).ParameterPath) | ConvertFrom-Json
+        $jsoncdba           = Get-Content -Path $($ARMObjects.Where({$_.ObjectType -eq "CosmosDBAccount"}).ParameterPath) | ConvertFrom-Json
+
+        $AzKeyVaultName       = $jsonKeyVault.parameters.name.value                         # Name of the Azure Keyvault
+        $AzStorageAccountName = $jsonStorageAccount.parameters.storageAccountName.value     # Name of the Azure Storage Account
+        $CosmosAccountName    = $jsoncdba.Parameters.Name.Value                             # Name of the Azure Cosmos Database Account
+
+        ## Azure Keyvault Secret Names
+        $AzStorageAccountKeyName      = "AzStorageAccountKey"       # Do not change
```

Comment:
Id suggest we update the comment to say something to the effect: "Unless you change the corresponding variable lookups in the azure function provisioning and reference implementation" or whatnot.  The extra pointers for someone who wants to customize their implementation couldnt hurt.

Reply:
We would need to provide a detailed step-by-step guidance on how to change, compile, and zip the functions then update the PowerShell script. This might be better included in a docs article.